### PR TITLE
[HDInsight] - Add scenario tests

### PR DESCRIPTION
**Pull request by idear1203**

### CODE REVIEW

File: hdinsight/resource-manager/v2018_06_01_preview/src/test/java/com/microsoft/azure/management/hdinsight/v2018_06_01_preview/utilities/Utilities.java [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+    public static String generateRandomClusterName(String prefix, int maxLen) {
+        return TestBase.generateRandomResourceName(prefix, maxLen);
+    }
+
+    public static Role findRoleByName(ClusterInner cluster, final String roleName) {
+        return Iterables.find(
+            cluster.properties()
+                .computeProfile()
+                .roles(),
+            new Predicate<Role>() {
```
Comment (wezhang) [inline hdinsight/resource-manager/v2018_06_01_preview/src/test/java/com/microsoft/azure/management/hdinsight/v2018_06_01_preview/utilities/Utilities.java]: The lambda function may work here for more simple

File: hdinsight/resource-manager/v2018_06_01_preview/src/test/java/com/microsoft/azure/management/hdinsight/v2018_06_01_preview/utilities/HDInsightTestResourceManager.java [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+    private final KeyVaultClient keyVaultClient;
+
+    private final String tenantId;
+
+    public HDInsightTestResourceManager(RestClient restClient, String subscription, String tenant) {
+        resourceManager = ResourceManager.authenticate(restClient).withSubscription(subscription);
+        storageManager = StorageManager.authenticate(restClient, subscription);
+        msiManager = MSIManager.authenticate(restClient, subscription);
+        keyVaultManager = KeyVaultManager.authenticate(restClient, tenant, subscription);
+        keyVaultClient = new KeyVaultClient(createKeyVaultRestClient(restClient));
```
Comment (wezhang) [inline hdinsight/resource-manager/v2018_06_01_preview/src/test/java/com/microsoft/azure/management/hdinsight/v2018_06_01_preview/utilities/HDInsightTestResourceManager.java]: Does the client access the real KeyVault service?

File: hdinsight/resource-manager/v2018_06_01_preview/src/test/java/com/microsoft/azure/management/hdinsight/v2018_06_01_preview/utilities/HDInsightTestResourceManager.java [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+        return createStorageAccount(resourceGroupName, STORAGE_ACCOUNT_NAME_PREFIX);
+    }
+
+    public com.microsoft.azure.management.storage.StorageAccount createStorageAccount(
+        String resourceGroupName,
+        String namePrefix) {
+        return createStorageAccount(resourceGroupName, namePrefix, STORAGE);
+    }
+
+    public com.microsoft.azure.management.storage.StorageAccount createStorageAccount(
```
Comment (wezhang) [inline hdinsight/resource-manager/v2018_06_01_preview/src/test/java/com/microsoft/azure/management/hdinsight/v2018_06_01_preview/utilities/HDInsightTestResourceManager.java]:
```suggestion
    public StorageAccount createStorageAccount(
```

File: hdinsight/resource-manager/v2018_06_01_preview/src/test/java/com/microsoft/azure/management/hdinsight/v2018_06_01_preview/utilities/HDInsightTestResourceManager.java [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+        String resourceGroupName,
```
Comment (wezhang) [inline hdinsight/resource-manager/v2018_06_01_preview/src/test/java/com/microsoft/azure/management/hdinsight/v2018_06_01_preview/utilities/HDInsightTestResourceManager.java]: The style looks weird. 😄

File: hdinsight/resource-manager/v2018_06_01_preview/src/test/java/com/microsoft/azure/management/hdinsight/v2018_06_01_preview/utilities/Utilities.java [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+
+import java.util.List;
+import java.util.Map;
+
+import static com.microsoft.azure.management.hdinsight.v2018_06_01_preview.scenariotests.HDInsightManagementTestBase.*;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class Utilities {
+
+    public static void validateCluster(String clusterName, ClusterCreateParametersExtended parameters, ClusterInner cluster) {
```
[the remark this answers is not in the document]
Comment (idear1203) [reply hdinsight/resource-manager/v2018_06_01_preview/src/test/java/com/microsoft/azure/management/hdinsight/v2018_06_01_preview/utilities/Utilities.java]: Good point! I will change those variable names.

File: hdinsight/resource-manager/v2018_06_01_preview/src/test/java/com/microsoft/azure/management/hdinsight/v2018_06_01_preview/utilities/Utilities.java (hunk 1 above)
Comment (idear1203) [reply hdinsight/resource-manager/v2018_06_01_preview/src/test/java/com/microsoft/azure/management/hdinsight/v2018_06_01_preview/utilities/Utilities.java]: Lambda expression is not available in Java 7. So I have to use anonymous class instead.

File: hdinsight/resource-manager/v2018_06_01_preview/src/test/java/com/microsoft/azure/management/hdinsight/v2018_06_01_preview/utilities/HDInsightTestResourceManager.java (hunk 2 above)
Comment (idear1203) [reply hdinsight/resource-manager/v2018_06_01_preview/src/test/java/com/microsoft/azure/management/hdinsight/v2018_06_01_preview/utilities/HDInsightTestResourceManager.java]: Yes, in record mode, the client access the real KeyVault service to get HTTP response.

File: hdinsight/resource-manager/v2018_06_01_preview/src/test/java/com/microsoft/azure/management/hdinsight/v2018_06_01_preview/utilities/HDInsightTestResourceManager.java (hunk 3 above)
Comment (idear1203) [reply hdinsight/resource-manager/v2018_06_01_preview/src/test/java/com/microsoft/azure/management/hdinsight/v2018_06_01_preview/utilities/HDInsightTestResourceManager.java]: Good catch! Other occurrences in this file will also changed.

File: hdinsight/resource-manager/v2018_06_01_preview/src/test/java/com/microsoft/azure/management/hdinsight/v2018_06_01_preview/utilities/HDInsightTestResourceManager.java (hunk 4 above)
Comment (idear1203) [reply hdinsight/resource-manager/v2018_06_01_preview/src/test/java/com/microsoft/azure/management/hdinsight/v2018_06_01_preview/utilities/HDInsightTestResourceManager.java]: Hmmm. I will join parameter list into one line for a few methods with short parameter list. However, for methods with long parameter list, I think this style has better readability.

**Pull request merged**